pkg/webhook/controlplane/ensurer.go

```diff
@@ -521,21 +517,41 @@ func (e *ensurer) addAdditionalFilesForResolvConfOptions(options []string, new *
 		permissions int32 = 0755
 		template          = `#!/bin/sh
 
-file=/run/systemd/resolve/resolv.conf
-tmp=/etc/resolv.conf.new
-dest=/etc/resolv.conf
+tmp=/etc/resolv-for-kubelet.conf.new
+dest=/etc/resolv-for-kubelet.conf
 line=%q
 
-if [ -f "$file" ]; then
-  cp "$file" "$tmp"
-  echo "" >> "$tmp"
-  echo "# updated by update-resolv-conf.service (installed by gardener-extension-provider-openstack)" >> "$tmp"
-  echo "$line" >> "$tmp"
-  mv "$tmp" "$dest"
-fi`
+is_systemd_resolved_system()
+{
+    if [ -f /run/systemd/resolve/resolv.conf ]; then
+      return 0
+    else
+      return 1
+    fi
+}
+
+rm -f "$tmp"
+if is_systemd_resolved_system; then
```

Comment:
Why not using `[ -f /run/systemd/resolve/resolv.conf ]` as condition here directly?
Is it the function `is_systemd_resolved_system()` somewhere as used?

Reply:
It was suggested by my other reviewer (see https://github.com/gardener/gardener-extension-provider-openstack/pull/375#discussion_r782951877). 
And I like it as it makes the semantics more visible.